RadixWallet/Clients/FactorSourcesClient/FactorSourcesClient+Interface.swift

```diff
@@ -173,7 +185,26 @@ public struct SigningFactor: Sendable, Hashable, Identifiable {
 	}
 }
 
+extension DeviceFactorSource {
+	func removingMainFlag() -> Self {
+		var copy = self
+		copy.common.flags.remove(.main)
+		return copy
+	}
+}
+
 extension FactorSourcesClient {
+	public func saveNewMainBDFS(_ newMainBDFS: NewMainBDFS) async throws {
```

Comment:
called at end of restore wallet from snapshot, when the profile snapshot has been saved.